tests/unit/theme/test_theme_jinja2filters.py

```diff
@@ -910,14 +913,15 @@ def test_publication_info_with_pub_info_and_conf_info(g_s, g):
     assert expected == result
 
 
-@mock.patch('inspirehep.modules.theme.jinja2filters.PersistentIdentifier.get')
-@mock.patch('inspirehep.modules.theme.jinja2filters.es.get_source')
-def test_publication_info_from_conference_recid_and_not_parent_recid(g_s, g):
+@mock.patch('inspirehep.modules.theme.jinja2filters.replace_refs')
+def test_publication_info_from_conference_recid_and_not_parent_recid(r_r):
     with_title = Record({
         'title': '20th International Workshop on Deep-Inelastic Scattering and Related Subjects'
     })
 
-    g_s.return_value = with_title
+    r_r.side_effect = lambda o, s: dict([('control_number',
+                                          o['$ref'].split('/')[-1])] +
+                                        with_title.items())
```

Comment:
That said, this is a little more logic in a mock than what I'm comfortable with: http://googletesting.blogspot.ch/2014/07/testing-on-toilet-dont-put-logic-in.html. Can you rewrite this more explicitly, maybe at the cost of boilerplate/repetition?